webrtc.html

```diff
@@ -1549,81 +1553,106 @@ <h4>Set the RTCSessionDescription</h4>
-                        set <var>transceiver</var>'s <a>[[\CurrentDirection]]</a> slot
-                        to an <code><a>RTCRtpTransceiverDirection</a></code>
-                        value representing the direction of the corresponding
-                        <a>media description</a>.</p>
+                        <p>Let <var>direction</var> be an <code>
+                        <a>RTCRtpTransceiverDirection</a></code> value
+                        representing the direction from the <a>media
+                        description</a>.</p>
                       </li>
                       <li>
-                        <p>
-                          If the <a>media description</a> is indicated as using
-                          an existing <a>media transport</a> according to
-                          [[!BUNDLE]], let <var>transport</var> and
-                          <var>rtcpTransport</var> be the
-                          <code><a>RTCDtlsTransport</a></code> objects
-                          representing the RTP and RTCP components of that
-                          transport, respectively.
-                        </p>
-                      </li>
-                      <li>
-                        <p>
-                          Otherwise, let <var>transport</var> and
-                          <var>rtcpTransport</var> be newly created
-                          <code><a>RTCDtlsTransport</a></code> objects, each
-                          with a new underlying
-                          <code><a>RTCIceTransport</a></code>. Though if RTCP
-                          multiplexing is negotiated according to [[!RFC5761]],
-                          or if <var>connection</var>'s
-                          <code><a>RTCRtcpMuxPolicy</a></code> is <code><a
-                          data-link-for="RTCRtcpMuxPolicy">require</a></code>,
-                          do not create any RTCP-specific transport objects,
-                          and instead let <var>rtcpTransport</var> equal
-                          <var>transport</var>.
-                        </p>
-                      </li>
-                      <li>
-                        <p>
-                          Set <var>transceiver</var>.<a>[[\Sender]]</a>.<a>[[\SenderTransport]]</a>
-                          to <var>transport</var>.
-                        </p>
-                      </li>
-                      <li>
-                        <p>
-                          Set <var>transceiver</var>.<a>[[\Sender]]</a>.<a>[[\SenderRtcpTransport]]</a>
-                          to <var>rtcpTransport</var>.
-                        </p>
+                        <p>If <var>direction</var> is <code>"sendrecv"</code> or
+                        <code>"recvonly"</code>,
+                        set <var>transceiver</var>'s <a>[[\Receptive]]</a> slot
+                        to <code>true</code>, otherwise set it to <code>false</code>.</p>
```

Comment:
Since this is in SLD, do we need to handle rollback?

I'm a bit concerned subsequent SLD calls with new offers can now mess with track state from a successful negotiation. E.g. The following is no longer benign:
```js
await pc.setRemoteDescription(incomingOfferSansTrack);
await pc.setLocalDescription(await pc.createAnswer());
signalingChannel.send({description: pc.localDescription});
// receiver.track.muted is true here, and can't be unmuted by any stray incoming rtp packets
pc.direction = "sendrecv";
await pc.setLocalDescription(await pc.createOffer());
await pc.setLocalDescription({type: "rollback"});
// stray incoming rtp packets can now cause unmute to fire on receiver.track 
```
Unsure what the way out here is.

Reply:
I guess the direction of the m-line in the remoteDescription is either 'inactive' or 'receiveonly' (meaning that the other endpoint does not intend to send)?

Then you change the direction (should be transceiver.direction, not pc.) indicating that you're willing to receive, and apply that offer locally before rolling back.

Would not the rollback essentially apply a description that (direction) says you're not willing to receive, and thus [[Receptive]] would go back to false? I'm not sure this PR makes this problem any worse.

Reply:
Ah. I opened #1783 cause I had the feeling you should in fact do the steps in 4.4.1.6, only with the previous description. That may be wrong, but it seems strange that SLD(offer) can change CurrentDirection, but a following SLD(rollback) would not change it back. Perhaps we should expand the text you mention to include [[Receptive]] but then also perhaps add [[CurrentDirection]].